File: Source/Core/Args.py
def HandleArgs(Arg):
  match Arg[1]:
    case "install" | "fetch" | "get":
      if len(Arg) <= 2:
        print("Error: No package(s) specified")
        exit(1)
      else:
        from Core import Install
        Install.QuikResolve(Arg[2:])
    case "help":
        from Commands import HandleHelp
        HandleHelp.HandleHelp()
    case "info":
      if len(Arg) > 3:
        print("Eror: can check info of only one package!")
        exit(1)
      if len(Arg) <= 2:
        print("Error: no packages provided!")
        exit(1) 
      from Core import Info
      Info.QueryInfo(Arg[2])
    case _:
      print(f"Unknown option {Arg[1]}")
      exit(1)

if __name__ == '__main__':
  print("This is not meant to be run!")
  exit(1)

File: Source/Core/Args.cpp
#include <Core/Args.h>
#include <Core/Install.h>
#include <Commands/HandleHelp.h>
#include <iostream>
#include <unordered_map>
#include <string>
#include <functional>
#include <stdlib.h>

void HandleInstall(int ArgLen, char *Arg[]) {
  if(ArgLen <= 2) {
    std::cout << "Error: No package(s) specified" << std::endl;
    exit(1);
  } else { 
    QuikResolve(ArgLen, Arg);
  }
}

int HandleArgs(int ArgLen, char *Arg[]) {
  std::unordered_map<std::string, std::function<void(int, char*[])>> Options = {
    {"install", [](int ArgLen, char *Arg[]) { HandleInstall(ArgLen, Arg); }},
    {"fetch", [](int ArgLen, char *Arg[]) { HandleInstall(ArgLen, Arg); }},
    {"get", [](int ArgLen, char *Arg[]) { HandleInstall(ArgLen, Arg); }},
    {"help", [](int ArgLen, char *Arg[]) { HandleHelp(); }}
  };
  auto it = Options.find(Arg[1]);
  if (it != Options.end()) {
    it->second(ArgLen, Arg);
  } else {
    std::cout << "Unknown option: " << Arg[1] << std::endl;
    exit(1);
  }
  return 0;
}
